### PR TITLE
feat: bnode relabelling param

### PR DESCRIPTION
**Pull request by jeswr**

### CODE REVIEW

**Review by phochste — CHANGES_REQUESTED** [review]
The option is `--no-bnode-relabeling` not `--no-bnode-relabelling`

**Comment by phochste** [comment]
With the following changes to `data/blogic.ts` the unit test for blogic should work:

```
export const data = `
@prefix : <urn:example:> .
@prefix log: <http://www.w3.org/2000/10/swap/log#> .

:Alice a :Person .

(_:x) log:onNegativeSurface {
   _:x a :Person .
   () log:onNegativeSurface {
       _:x a :Human .
   } .
} .

(_:x) log:onNegativeSurface {
   _:x a :Human .
   () log:onNegativeAnswerSurface {
      _:x a :Human.
   }
} .
`;

export const result = `
@prefix : <urn:example:>.
@prefix log: <http://www.w3.org/2000/10/swap/log#>.

<urn:example:Alice> a <urn:example:Human>.
`;
```

File: __test_utils__/util.ts [hunk 1]
```diff
@@ -276,5 +278,13 @@ export function universalTests() {
...
       await expect(n3reasoner('invalid', 'invalid')).rejects.toThrowError('Error while executing query');
     });
+
+    it('should execute the n3reasoner on surface query', async () => {
+      await expect(n3reasoner(surfaceQueryQuads)).rejects.toThrow(/inference_fuse/);
+    });
+
+    it('should execute the n3reasoner on surface query [bnodeRelabeling: false]', async () => {
+      await expect(n3reasoner(surfaceQueryQuads, undefined, { bnodeRelabeling: false })).rejects.toThrow(/inference_fuse/);
+    });
```
Comment (jeswr) [inline __test_utils__/util.ts:288]: @phochste at the moment we are getting an inference fuse thrown regardless of whether or not `bnodeRelabeling` is enabled. Can you provide a test case that distinguishes between the two? Note the `--pass-only-new` flag is also enabled in this test due to https://github.com/eyereasoner/eye-js/blob/df8aa4bb220fdd0799ca045c6218a5c48ba895bb/lib/transformers.ts#L93.

**Comment by phochste** [comment]
The inference fuse can be tested with :

```
@prefix : <https://example.org/ns#> .
@prefix log: <http://www.w3.org/2000/10/swap/log#> .

:ABC a :DepartmentPreference .
:ABC a :ResearcherPreference .

(_:WHAT) log:onNegativeSurface {
    _:WHAT a :DepartmentPreference .
    _:WHAT a :ResearcherPreference .
} .
```

With `bnodeRelabeling: true` it should fuse, without it should pass.

File: __test_utils__/util.ts [hunk 2]
```diff
@@ -276,5 +280,31 @@ export function universalTests() {
...
+      await expect(n3reasoner(surfaceRelabelingQueryQuads, undefined, { bnodeRelabeling: false })).rejects.toThrow(/inference_fuse/);
+    });
+
+    it('should execute the n3reasoner on surface query that doesnt fuse [bnodeRelabeling: true]', async () => {
+      await expect(n3reasoner(surfaceRelabelingQuery, undefined, { outputType: 'quads' })).resolves.toBeRdfIsomorphic(surfaceRelabelingResultQuads);
+      await expect(n3reasoner(surfaceRelabelingQuery, undefined, { bnodeRelabeling: true, outputType: 'quads' })).resolves.toBeRdfIsomorphic(surfaceRelabelingResultQuads);
+      await expect(n3reasoner(
+        surfaceRelabelingQueryQuads
+          // see https://github.com/rdfjs/N3.js/issues/332
+          .map(quad => mapTerms(quad, term => term.termType === 'BlankNode' ? DataFactory.blankNode(term.value.replace(/\./g, '__dot__')) : term))
```
Comment (jeswr) [inline __test_utils__/util.ts:306]: We should get rid of this once https://github.com/rdfjs/N3.js/issues/332 is resolved.

**Pull request merged**